src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -92,17 +96,23 @@ def _try_write_trace_to_cosmosdb(all_spans, logger: logging.Logger):
 
         get_client(CosmosDBContainerName.LINE_SUMMARY, subscription_id, resource_group_name, workspace_name)
 
+        # For local scenario, we already get created_by info in advance as starting the service.
+        # But if customer didn't run `az login`, we can't get it.
+        # So, we try to get and cache it again after getting CosmosDB token.
+        # Don't bother to run in new thread because in most cases, it will be cached.
+        created_by = get_created_by_info_with_cache() if get_created_by_info_with_cache else {}
```

Comment:
little confused, when will this line go to `{}` part? when local to cloud is not enabled?

Reply:
Make `get_created_by_info_with_cache` nullable is better to let this file's code reusage.
For now, client should always pass in `get_created_by_info_with_cache`

Reply:
Don't consider nullable `get_created_by_info_with_cache`. Since now we will call it in separate worker thread.
It introduces unnecessary complexity for code.